File: pset2/crack.c
/**
 * crack.c
 * 
 * Natalia Ostiak
 * Technology Nation
 * 
 * Cracks passwords.
 */
 
#define _XOPEN_SOURCE
#include <unistd.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>

// constant
const char *SALT = "50";

char key[5];
char *result;
int compareResult;


int main(int argc, string argv[])
{
    if (argc != 2)
    {
        printf("Usage:./crack hash\n");
        return 1;
    }
    
    char letters[53];
    int i, j, k, l;
    for (i = 0; i < 26; i++)
        letters[i] = (char) (i + 65);
    for (i = 0; i < 26; i++)   
        letters[i+26] = (char) (i + 97);
        
    // if the password is one character
    for (i = 0; i < 52; i++) 
    {
        key[0] = letters[i];
        result = crypt(key, SALT);
        compareResult = strcmp(result, argv[1]);
        if (compareResult == 0)
        {
            printf("%s\n", key);
            break;
        }
       
    }        
            
            
    // if the password is two characters
    for (i = 0; i < 52; i++) 
    {
        key[0] = letters[i];
        for (j = 0; j < 52; j++)
        { 
            key[1] = letters[j];
            result = crypt(key, SALT);
            compareResult = strcmp(result, argv[1]);
            if (compareResult == 0)
            {
                printf("%s\n", key);
                break;
            }
        }    
    }        

    // if the password is three characters
    for (i = 0; i < 52; i++) 
    {
        key[0] = letters[i];
        for (j = 0; j < 52; j++)
        {
            key[1] = letters[j];
            for (k = 0; k < 52; k++)
            {
                key[2] = letters[k];
                result = crypt(key, SALT);
                compareResult = strcmp(result, argv[1]);
                if (compareResult == 0)
                {
                    printf("%s\n", key);
                    break;
                }
            }
        }    
    }        

    // if the password is four characters
    for (i = 0; i < 52; i++) 
    {
        key[0] = letters[i];
        for (j = 0; j < 52; j++)
        {
            key[1] = letters[j];
            for (k = 0; k < 52; k++)
            {
                key[2] = letters[k];
                for (l = 0; l < 52; l++)
                {
                    key[3] = letters[l];
                    result = crypt(key, SALT);
                    compareResult = strcmp(result, argv[1]);
                    if (compareResult == 0)
                    {
                        printf("%s\n", key);
                        break;
                    }
                }
            }
        }    
    }        
   return 0;
}    

File: pset6/crack.py
# crack.py
# 
# Natalia Ostiak
# Technology Nation
# 
# Cracks passwords.
 
import sys
import crypt

salt = "50"
letters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz'

if len(sys.argv) != 2:
    print(" Usage:python crack.py hash")
    exit(1) 
    
# if the password is one character
for c in letters: 
    key = c
    result = crypt.crypt(key, salt)
    if result == sys.argv[1]:
        print("{}".format(key))
        exit(0)
       
# if the password is two characters
for c in letters: 
    key21 = c
    for c in letters:
        key22 = c
        key2 = key21 + key22
        result = crypt.crypt(key2, salt)
        if result == sys.argv[1]:
            print("{}".format(key2))
            exit(0)
                
# if the password is three characters
for c in letters: 
    key31 = c
    for c in letters:
        key32 = c
        for c in letters:
            key33 = c
            key3 = key31 + key32 + key33
            result = crypt.crypt(key3, salt)
            if result == sys.argv[1]:
                print("{}".format(key3))
                exit(0)
                   
# if the password is four characters
for c in letters: 
    key41 = c
    for c in letters:
        key42 = c
        for c in letters:
            key43 = c
            for c in letters:
                key44 = c
                key4 = key41 + key42 + key43 + key44
                result = crypt.crypt(key4, salt)
                if result == sys.argv[1]:
                    print("{}".format(key4))
                    exit(0) 
